### ISSUE TITLE
How do you output the causal graph once the model is trained?

### ISSUE THREAD

**Issue by parkererickson** [loeweX/AmortizedCausalDiscovery#2]
Hi, I am trying to understand how to extract the causal graph inferred by the model once it has been trained. Can you shed some light on how I would go about this?

Thanks!

**Comment by dmadras** [comment]
Hi Parker -

The outputted logits from the encoder (plus a softmax) are the causal graph estimated edge probabilities.

David
________________________________

**Comment by parkererickson** [comment]
Great, thanks!

**Comment by parkererickson** [comment]
I got the output working - it appears to be a 20 X 2 matrix for each simulation. These represent the edges? Since I am running on the 5 springs dataset, I am assuming that each row is a specific edge between two vertices in the graph, but I am unclear how the pairs are ordered in relation to the adjacency matrix. Also, does the first column represent that an edge exists (1 if true, 0 if false) or is it some other scheme?  Thanks again!

**Comment by dmadras** [comment]
1.   The rel_rec and rel_send variables should define which arrows in the graph correspond to the 20 entries in the vector ("relations receiving" and "relations sending").
  2.  The 0 index refers to no edge - see for instance codebase/model/MLPDecoder line 59: the (default) option is to skip the edges where the 0 index is chosen when doing message-passing.

________________________________

**Comment by parkererickson** [comment]
Thanks again!
